packages/adapter/adapter-node/src/common.ts

```diff
@@ -1,135 +0,0 @@
-import type { AdapterRequestContext, HattipHandler } from "@hattip/core";
-import {
-	createServer as createHttpServer,
-	Server as HttpServer,
-	IncomingMessage,
-	ServerResponse,
-	ServerOptions,
-} from "node:http";
-import type { Socket } from "node:net";
-import { NodeRequestAdapterOptions, createRequestAdapter } from "./request";
-import { sendResponse } from "./response";
-import process from "node:process";
-
-interface PossiblyEncryptedSocket extends Socket {
-	encrypted?: boolean;
-}
-
-/**
- * `IncomingMessage` possibly augmented by Express-specific
- * `ip` and `protocol` properties.
- */
-export interface DecoratedRequest extends Omit<IncomingMessage, "socket"> {
-	ip?: string;
-	protocol?: string;
-	socket: PossiblyEncryptedSocket;
```

Comment:
This property was non-optional, but I saw another `DecoratedRequest` type in `src/request.ts` had it set as optional. Which one is right?

Reply:
It's not optional, `req.socket` is always defined.

What I'm trying to achieve is that there's a [badly documented](https://github.com/nodejs/node/issues/41863) `req.socket.encrypted` property that only exists when using `node:https`. The adapter uses to infer the protocol (https or http) part of the URL when origin isn't manually set and `trustProxy` is false. (`req.protocol` is a non-standard Express property, useful embedding a Hattip app into an Express app as a middleware).

I think the one in `src/request.ts` is just an error on my part.